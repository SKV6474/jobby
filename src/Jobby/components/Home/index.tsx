import React from "react";

import { JOB_ROUTE } from "../../constants/Path";
import { Props } from "../../interface";

import {
  ContentContainer,
  DescriptionContainer,
  HomeContainer,
  Link,
  TitleContainer,
} from "./styledComponents";

const Home = (props: Props) => {
  return (
    <HomeContainer>
      <ContentContainer>
        <TitleContainer>Find The Job That Fits Your Life</TitleContainer>
        <DescriptionContainer>
          Millions of people are searching for jobs, salary information, company
          reviews. Find the job that fits your abilities and potential.
        </DescriptionContainer>
        <div>
          <Link to={JOB_ROUTE}>Find Jobs</Link>
        </div>
      </ContentContainer>
    </HomeContainer>
  );
};

export default Home;
