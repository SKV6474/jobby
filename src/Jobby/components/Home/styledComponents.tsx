import { NavLink } from "react-router-dom";
import styled from "styled-components";

export const HomeContainer = styled.div`
  background-image: url("https://assets.ccbp.in/frontend/react-js/home-lg-bg.png");
  background-position: 50% 50%;
  height: calc(100vh - 90px);
  background-size: cover;
  background-repeat: no-repeat;
  padding: 0px 40px 0 50px;
  @media screen and (max-width: 768px) {
    background-image: url("https://assets.ccbp.in/frontend/react-js/home-sm-bg.png");
    padding: 20px 20px 0px;
  }
`;

export const ContentContainer = styled.div`
  display: flex;
  justify-content: center;
  flex-direction: column;
  box-sizing: border-box;
  width: 650px;
  height: 100%;
  color: #fff;
  font-family: Arial, Helvetica, sans-serif;
  @media screen and (max-width: 768px) {
    width: calc(100vw - 40px);
    justify-content: flex-start;
  }
`;

export const TitleContainer = styled.div`
  font-size: 70px;
  font-weight: 600;
  line-height: 1.25;
  margin-bottom: 30px;
  @media screen and (max-width: 768px) {
    font-size: 28px;
  }
`;

export const DescriptionContainer = styled.div`
  font-size: 28px;
  line-height: 1.25;
  margin-bottom: 50px;
  @media screen and (max-width: 768px) {
    font-size: 18px;
  }
`;

export const Link = styled(NavLink)`
  text-decoration: none;
  color: #fff;
  background-color: #6366f1;
  padding: 15px 20px;
  border-radius: 5px;
  font-weight: 600;
  @media screen and (max-width: 768px) {
    padding: 10px 15px;
  }
`;
