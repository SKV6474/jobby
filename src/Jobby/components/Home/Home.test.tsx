import { JSDOM } from "jsdom";
import { BrowserRouter } from "react-router-dom";

import renderer from "react-test-renderer";
import Home from ".";

const dom = new JSDOM("<!doctype html><html><body></body></html>");
// global.window = dom.window;
global.document = dom.window.document;

test("renders correctly", () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <Home />
      </BrowserRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
