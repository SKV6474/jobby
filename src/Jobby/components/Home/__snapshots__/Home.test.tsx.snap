// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sc-beqWaB lkEORj"
>
  <div
    className="sc-gueYoa cjmCoc"
  >
    <div
      className="sc-dmqHEX fQGRlX"
    >
      Find The Job That Fits Your Life
    </div>
    <div
      className="sc-hLseeU dtcKob"
    >
      Millions of people are searching for jobs, salary information, company reviews. Find the job that fits your abilities and potential.
    </div>
    <div>
      <a
        aria-current={null}
        className="sc-eDDNvR jAwwND"
        href="/jobs"
        onClick={[Function]}
      >
        Find Jobs
      </a>
    </div>
  </div>
</div>
`;
